tests/run/i17555.scala

```diff
@@ -0,0 +1,17 @@
+class Root {
+  override def toString() = "Root"
+}
+trait A extends Root with B {  }
+trait B {
+   override def toString() = "B"
+}
+case class C() extends A {
+  override def toString() = super.toString()
+}
+class D() extends A, Serializable {
+  override def toString() = super.toString()
+}
+
+@main def Test =
+  println(C())
+  println(D())
```

Comment:
vulpix does not verify output without a check file, so `assert` is more effective or defensive.

Reply:
Oh, yes, that was an oversight. Thanks for pointing it out!